core/vm/jump_table.go

```diff
@@ -1052,19 +1106,26 @@ func newFrontierInstructionSet() JumpTable {
 			minStack:   minStack(2, 0),
 			maxStack:   maxStack(2, 0),
 			memorySize: memoryReturn,
+			terminal:   true,
 		},
 		SELFDESTRUCT: {
 			execute:    opSelfdestruct,
 			dynamicGas: gasSelfdestruct,
 			minStack:   minStack(1, 0),
 			maxStack:   maxStack(1, 0),
 		},
+		INVALID: {
+			execute:  opUndefined,
+			minStack: minStack(0, 0),
+			maxStack: maxStack(0, 0),
+			terminal: true,
+		},
 	}
 
 	// Fill all unassigned slots with opUndefined.
 	for i, entry := range tbl {
 		if entry == nil {
-			tbl[i] = &operation{execute: opUndefined, maxStack: maxStack(0, 0)}
+			tbl[i] = &operation{execute: opUndefined, maxStack: maxStack(0, 0), undefined: true}
```

Comment:
The way this is defined, makes it so that a code-container containing nothing but the `FE` (invalid) opcode is a valid container. 
```
$ go run ./cmd/eofdump/  eof --hex 0xef000101000402000100010400000000800000fe
OK
```
That is, it behaves just like `STOP`: 
```
[user@work go-ethereum]$ go run ./cmd/eofdump/  eof --hex 0xef00010100040200010001040000000080000000
OK
```
But different from an actual undefined opcode, like `EF`:
```
[user@work go-ethereum]$ go run ./cmd/eofdump/  eof --hex 0xef000101000402000100010400000000800000ef
err(15): undefined instruction
exit status 1
```
This seems wrong to me. @shemnon what do you say ?

Reply:
It appears that this is covered by at least one of the common consensus-tests for eof. It seems very weird that we have one "blessed" undefined which acts differently from the actual undefined. Afaik, the idea with the _designated_ undefined was that it will be left undefined, forever -- as opposed to non-designated undefineds, which may become used in the future.

Reply:
The spec does metion this:https://github.com/ipsilon/eof/blob/main/spec/eof.md#stack-validation . 

>- Terminating instructions:
>
>    - ending function execution: RETF, JUMPF,
>     - ending whole EVM execution: STOP, RETURN, RETURNCONTRACT, REVERT, INVALID.

I think this behaviour is utterly incomprehensible. Would love to know a bit more about the thoughts behind it. Why is it useful to allow `INVALID`? 

IMO doing so violates the agreement/idea behind `INVALID`, by turning it into a unique opcode with a special purpose, as opposed to "one in the general soup of invalid/undefined operations".

Reply:
Ok, thanks for the explanation. Re-reading what I wrote -- sorry if I sounded overly aggressive !

...  :cake: